### ISSUE TITLE
how to get the best configuration for cpu

### ISSUE THREAD

**Issue by developernew123** [xmrig/xmrig#461]
I want to get the best configuration for cpu 
I thought the more threads I use , the more hashrate I get 

but I found that this is wrong due to the cpu cache

is there any formula to get the best configuration (number of threads)

**Comment by BusaSpectre** [comment]
There are two approaches.

1) Cache size divided by 2 MB - this holds true for most processors.
2) Set threads to match actual core count (so don't count hyperthreading or total "logical" cores).

Examples:
i7-2600K - 4 threads - (4 cores, 8 threads) - ~290 H/s
i7-3770K - 4 threads - (4 cores, 8 threads) - ~300 H/s
R7-1700 - 8 threads - (8 cores, 16 threads) - ~555 H/s
i7-7820X - 8 threads - (8 cores, 16 threads) - ~595 H/s

I was surprised by the i7-7820X, as following the cache size divided by 2 MB would have been less than 8 threads.

In all of these instances, I don't use any additional settings or parameters.

I've found that manually setting core affinity causes the computer to bog or even temporarily freeze while performing other tasks. Even when not on other tasks, I've noticed higher and more consistent hash rate without affinity.

The only extra parameter I would recommend is priority. IF you are multitasking on the computer, setting the priority higher/lower could help the computer behave as you want.

TL;DR - Set threads to equal PHYSICAL cores.

**Comment by developernew123** [comment]
what about using --safe parameter , I think it's better

**Comment by Gill1000** [comment]
--safe parameter uses only 1 thread. (Default)....i know because i m using this.

**Comment by developernew123** [comment]
won't it increase threads even if the cache is 10 mb ?!!!!
